### ISSUE TITLE
Expandable DataGrid: Expand/Collapse state change is not read by screen reader

### ISSUE THREAD

**Issue by kalpeshhpatel** [vmware/clarity#4089]
## Describe the bug

For the data grid with expandable rows, when the row is expanded/collapsed, the state change is not read by screen-reader.

## How to reproduce

1) Go to: https://v2.clarity.design/datagrid/expandable-rows
2) Use the NVDA screen reader.
3) Expand/Collapse the row of data-grid.
4) Observe the NVDA speech-viewer.
5) The state change of row expand/collapse is not read by the screen reader.

## Expected behavior

Whenever row detail pane is expanded/collapsed, the screen reader should be able to read the state change.

## Versions

This issue is reproducible with the latest version of Clarity with all browsers.

<img width="1250" alt="Screen Shot 2019-12-05 at 12 19 55 PM" src="https://user-images.githubusercontent.com/5223493/70210802-a2dad500-1759-11ea-8f7c-c592c81c967b.png">

**Comment by hippee-lee** [comment]
This also happens for VoiceOver+Safari.
It looks like the expand/collapse button is missing `aria-expanded` attribute.

**Comment by chlane** [comment]
This issue is mentioned in issue 12 (The "datagrid-expandable-caret-button" button does not indicate its expanded state to assistive technologies) on http://cx-guidelines/clarityTesting/

and should have been addressed in

https://github.com/vmware/clarity/issues/3371

**Comment by gnomeontherun** [comment]
It was not addressed so we need to fix this one. Should be pretty straight forward to add the aria attributes.

**Comment by chlane** [comment]
Fix verified on https://clarity.design/documentation/datagrid/expandable-rows

**Comment by kalpeshhpatel** [comment]
The issue doesn't seems to be available for Clarity version 2.x.

Can you guys please confirm on this?

Thanks
